src/colmap/scene/synthetic.h

```diff
@@ -62,6 +62,6 @@ struct SyntheticDatasetOptions {
 
 void SynthesizeDataset(const SyntheticDatasetOptions& options,
                        Reconstruction* reconstruction,
-                       Database* database);
+                       Database* database = nullptr);
```

Comment:
nit: could have one minimal unit test that checks for non-crash when database is not passed.

Reply:
Done.